packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -115,8 +133,28 @@ export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
   return ReactDOM.createPortal(template, document.body);
 
   function handleSelection(selection: ComboboxOption) {
-    props.onSelect(selection);
-    setSearchValue("");
-    setOpen(false);
+    const callbackHandler = props.onSelect
+      ? props.onSelect
+      : setInternalSelected;
+
+    if (multiselect) {
+      handleMultiSelect(callbackHandler, selectedOptions, selection);
+    } else {
+      callbackHandler([selection]);
+      setSearchValue("");
+      setOpen(false);
+    }
```

Comment:
I don't know why I made a function for multi select and not single select when they're both like 3 lines. maybe I thought multi was going to be larger than it ended up being

open to changing this in whichever direction

Reply:
toss handleSingleSelect into a function if you're down for it. Might only be 3 lines now, but it could grow in the future and it'll be less anxiety-inducing to add a line in the future than refactor the whole thing.